src/materials/DamagePlasticityStressUpdate.C

```diff
@@ -532,66 +512,91 @@ DamagePlasticityStressUpdate::setIntnlDerivativesV(const std::vector<Real> & tri
 }
 
 Real
-DamagePlasticityStressUpdate::ft(const std::vector<Real> & intnl) const
+DamagePlasticityStressUpdate::fbar(const Real & f0,
+                                   const Real & a,
+                                   const Real & exponent,
+                                   const Real & kappa) const
 {
-  Real sqrtPhi_t = std::sqrt(1. + _at * (2. + _at) * intnl[0]);
-  if (_zt > sqrtPhi_t / _at)
-    return _ft0 * std::pow(_zt - sqrtPhi_t / _at, (1. - _dt_bt)) * sqrtPhi_t;
-  else
-    return _ft0 * 1.E-6; // A very small number (instead of zero) is used for end of softening
+  Real phi = 1. + a * (2. + a) * kappa;
+  Real sqrt_phi = std::sqrt(phi);
+  Real v = sqrt_phi;
+  Real u = (1 + a) / a - sqrt_phi / a;
+  Real cal_fbar = f0 * std::pow(u, exponent) * v;
+  return (u > 0) ? cal_fbar : 1.E-6;
```

Comment:
Leave in comment explaining what 1e-6 is

Reply:
Addressed